src/main/java/uk/gov/hmcts/reform/coh/controller/OnlineHearingController.java

```diff
@@ -108,6 +110,10 @@ public OnlineHearingsResponse retrieveOnlineHearings(@RequestParam("case_id") Li
     @PostMapping(value = "", consumes = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity createOnlineHearing(@RequestBody OnlineHearingRequest body) {
 
+        if (!onlineHearingService.retrieveOnlineHearingByCaseIds(Arrays.asList(body.getCaseId())).isEmpty()) {
+             return new ResponseEntity<>("Duplicate case found", HttpStatus.CONFLICT);
```

Comment:
I think a unit test to check the response if duplicate is found would be good.

Reply:
Done